x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/split_field/by_field.tsx

```diff
@@ -22,7 +25,11 @@ export const ByFieldSelector: FC<Props> = ({ detectorIndex }) => {
   const { jobCreator: jc, jobCreatorUpdate, jobCreatorUpdated } = useContext(JobCreatorContext);
   const jobCreator = jc as PopulationJobCreator;
 
-  const { categoryFields: allCategoryFields } = newJobCapsService;
+  const [runtimeCategoryFields] = useState(filterCategoryFields(jobCreator.runtimeFields));
+  const [allCategoryFields] = useState([
+    ...newJobCapsService.categoryFields,
+    ...runtimeCategoryFields,
+  ]);
```

Comment:
Does this need `useState` if a `setFields` is never needed? Maybe it doesn't need to be wrapped in `useState` at all or can make use of `useMemo`?

Reply:
updated in https://github.com/elastic/kibana/pull/91168/commits/3e19ad9587a3e70040dcf4523c0e377a0bd54ae2